File: solution/783.search-in-a-binary-search-tree/search-in-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
        if (!root) {
          return null;
        }
        let temp = root;
        while (temp) {
          if (temp.val == val) {
            break;
          } else if (temp.val > val) {
            temp = temp.left;
          } else {
            temp = temp.right;
          }
        }
        return temp;
      };

File: solution/700.search-in-a-binary-search-tree/search-in-a-binary-search-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* searchBST(struct TreeNode* root, int val) {
    if(!root){
        return NULL;
    }
    struct TreeNode * temp = root;
    while(temp){
        if(temp->val == val){
            break;
        }else if(temp->val > val){
            temp = temp->left;
        }else {
            temp = temp->right;
        }
    }
    return temp;
}